### PR TITLE
7100 - validate that the alteration type is benefit company

### PR DESCRIPTION
**Pull request by jordiwes**
*Issue #:* /bcgov/entity#7100

*Description of changes:*
add validation that the legal type is benefit company

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

**Comment by jordiwes** [comment]
Do I also have to validate the name request legal type?

File: legal-api/src/legal_api/services/filings/validations/alteration.py [hunk 1]
```diff
@@ -97,3 +98,14 @@ def company_name_validation(filing):
                         'path': legal_name_path})
 
     return msg
+
+
+def type_change_validation(filing):
+    """Validate type change."""
+    msg = []
+    legal_type_path: Final = '/filing/business/legalType'
```
Comment (thorwolpert) [inline legal-api/src/legal_api/services/filings/validations/alteration.py:106]: There are 2 sections that have ```legalType```
**Metadata**: existing business info
```/filing/business/legalType```
and the **filing** data
```filing/alteration/business/legalType```
is this checking the right one?

**Review by thorwolpert — COMMENTED** [review]
Is this checking the current state, or the desired state?

File: legal-api/src/legal_api/services/filings/validations/alteration.py (hunk 1 above)
Comment (jordiwes) [reply legal-api/src/legal_api/services/filings/validations/alteration.py]: I changed to use the alteration path

File: legal-api/src/legal_api/services/filings/validations/alteration.py [hunk 2]
```diff
@@ -97,3 +98,15 @@ def company_name_validation(filing):
...
     return msg
+
+
+def type_change_validation(filing):
+    """Validate type change."""
+    msg = []
+    legal_type_path: Final = '/filing/alteration/business/legalType'
+    # you must alter to a bc benefit company
+    if get_str(filing, legal_type_path) != 'BEN':
+        msg.append({'error': babel('Your business type has not been updated to a BC Benefit Company.'),
```
Comment (severinbeauvais) [inline legal-api/src/legal_api/services/filings/validations/alteration.py:109]: Usually it's good to say why the error occurred (esp as we display the error text in the UI). Is this message approved by PO/UX?

Comment (jordiwes) [reply legal-api/src/legal_api/services/filings/validations/alteration.py:109]: It's from the ticket: https://app.zenhub.com/workspaces/entity-5bf2f2164b5806bc2bf60531/issues/bcgov/entity/7100 :)

**Review by severinbeauvais — APPROVED** [review]

**Review by thorwolpert — APPROVED** [review]

**Pull request merged**